include/rocksdb/env.h

```diff
@@ -540,6 +540,9 @@ class Env {
     return Status::NotSupported();
   }
 
+  // Check whether the specified path is a directory
+  virtual Status IsDirectory(const std::string& path, bool* is_dir) = 0;
```

Comment:
I wonder if we should have a default implementation that returns ```Status::NotSupported()```, rather than pure abstract, in order to avoid disruption. Especially since RocksDB itself doesn't need to use it at the moment.

Reply:
In the future, I am thinking whether it's possible to implement these functions as
```
GetFileSystem()->XXX();
```
For now, I will just return `NotSupported()`.

Reply:
Yes, they could be redirected to the ```FileSystem``` implementation.